notebooks/Binary Classification with the UCI Credit Card Payment Default Dataset.ipynb

```diff
@@ -0,0 +1,1150 @@
K/5GYZhGMZpEbQjv/zAhK0NwzCKJ8U6+akJWxuGYRRLbNozn/AIW+f0AbjrrruoXr06TZs29bZds2YNbdu2pVmzZnTr1o1Dhw5l6Hv79u1UqFCB5557zls2atQoateuneNL50uWLKFVq1Y0a9aMVq1asWzZslzbHzt2jNtvv50GDRpw2WWXkZSU9Hcei2EYRpHEkt8Zpn///lmkwgYNGsSECRNYu3YtN910UwYHBoAHHnggg9QZQLdu3Vi5cmWO16patSofffQRa9euZebMmfTt2zfX9m+99RaVK1dmy5YtPPDAAzzyyCN5vUXDMIwiT9Amv4J0dcjUf7yILMqvuK+++mrvO3oeEhMTufrqqwFo3759BnHqBQsWUK9ePSIiIjK0adOmDTVq1MjxWi1atKBmTUePOyIigqNHj3Ls2LEc2y9cuJB+/foB0LNnT5YuXep5/cMwDOOsIQUA7RAAACAASURBVGiTHwXr6uDbf5Sqdi+YW3Bo2rQpixY5+XXevHleceqUlBSeeeYZnnzyyb99jfnz59OiRQtKly6dY71du3ZRu3ZtwLFeqlixIgcOHPjb1zcMwyhKnC0bXr7CfS/Px9XhWmAREAOOq4OI3IXj6tCVHFwdAiWvwtYeQdfff/+dlJQU7/F9993HuHHjGDlyJFdccQUhISHExsby2muv0aFDB+Li4khKSqJs2bJZRGHT09NzFYrdtm0bTzzxBBMnTvTb3lds9vDhw3z33Xdem6SjR4/yzTffULFixVyfR2FQlIRy80owxw7BHb/FXjgUqdgL4836/PjgKrzgJPCFwP3u8R3AW+73b4GWPm1qAjuB5QH0nwbEAd8DN+ZWv/ZF9bXOIx/n+PGwbds2jYiIUH8kJibqJZdcoqqqV155pdapU0fr1KmjFStW1MqVK+vkyZMz1C9fvrzffjzs2LFDL774Yv3666/9ni9fvnwG1YUOHTrot99+q6qqJ06c0PPOO09PnjyZ4zUKk7NN7SKYCOb4LfbCwRRe8ocCdXUALnTr1wOWichaVf0l/8I/xd69e6levTonT55k3LhxXn1Nj28fQExMDBUqVGDo0KEB95ucnEzXrl0ZP348V1xxRUBtunfvzsyZM2nbti3vv/8+7dq18xrcGoZhnC2cDWt+BeLqoK5/n6puxdEMbZEfQffu3Zu2bduSmJhIWFgYb731FrNnz6Zhw4aEh4dTs2ZNBgwYkGs/Dz/8MGFhYaSmphIWFkZMTAzgCFqPGTMGgFdeeYUtW7bw1FNPERUVRVRUlNcFwrf9rbfe6m0/cOBADhw4QIMGDXjhhReYMGFCfty2YRhG0aIwhpv58cG/sPW9wOuZyr4ErvI5nkHuwtaVgdLu96rAZqBJTm1M2LpwsNgLj2CO32IvHIrStGcwj/z80Rv4MFPZfOAfeeynMRAnImuA5cAEVd2QD/EZhmEYRYCgXfPTgnV1+BZolls9wzAMIzg520Z+hmEYhpErQTvyyw/M1cEwDKN4UqyTn5qrg2EYRrHEpj3zidxcHTz4c3WIj4+nTZs2REVF0bp1a6/g9IEDB7j22muzvN/3119/eV9diIqKomrVqgwfPtxvXOPHj6dBgwY0atSIzz77zFveqVMnmjdvTkREBC+88ALp6emAuToYhlE8sOR3hvHn6vDwww/z5JNPEh8fz9ixY3n44YcBKFOmDE899VQGKyOAc845h/j4eO+nTp063HzzzVmutWHDBubMmcP69etZvHgxgwcP9ia5uXPnsmbNGtatW0dycjLz5s0DzNXBMIziQdAmv4J2dXDrnCsiu0TklfyK25+rg4h4Pfz+/PNPrxND+fLlufLKKylTpky2/W3evJm9e/dy1VVXZTm3cOFCevXqRenSpbnoooto0KCBd1R57rnnApCWlkZaWppXxcVcHQzDKA4EbfKj4F0dAJ7CeUm+QHnppZcYOXIktWvXZsSIEYwfPz7gtrNnz+b222/3K0Hm69AAEBYWxq5du7zHHTt2pHr16pQtW5aePXtmaWOuDoZhnK2cLRte8t3VQURaAecDi4HW2dQJ2NXBV8k8s6vDyy+/zMCBA7nmmmtYvnw5N998M88//7y3/saNG9m1a5dfNfRp06bx2GOP+T23c+dOfv75Z++53bt3s379eqpWrQrAY489xvHjx4mJieHFF1+kdevW5upwBgnm2CG447fYC4eiFHvQJz8RKQF0xklSADcCi1V1k4j8ISItVfVH99xwYCWwWVUzv+Lg22cI8DzQF7guu3qq+gbwBsCF9Rro82uzf5xJfaJPfU9Konz58kRHO2U9evRg/vz5iAjXXHMNL774ovecp/7hw4czlAGsWbOGUqVKce+99/q95nfffQfgbTd+/Hg6dOhA27ZtM9RbtmwZSUlJjBgxgkaNGhEWFkbbtm1JS0vj2LFjdO/evciKW8fGxmZ5LsFCMMcOwR2/xV44FKXYg3na0+PqEAdsJ6Orwxz3u8fVAfCKVS8DXsul78HAp6q6I18jzoaaNWvy5ZfO7OqyZcu4+OKLA2o3e/Zsevfune357t27M2fOHI4dO8a2bdvYvHkzl156KYcPH2b37t2As+b3ww8/EB4e7m0zc+ZMAHN1MAzjrCWYR35HVDXDO3o+rg5NRUSBUEBF5GE9tWsjEFeHtsBVIjIYqACUEpHDqvro3w26d+/exMbGsn//fsLCwvj3v//Nf//7X4YNG0ZaWhplypThjTfe8NavW7cuhw4d4vjx4yxYsIDPP/+cJk2aAM6OzU8//TRD/4sWLSIuLo6xY8cSERHBbbfdRpMmTShRogRTpkwhNDSUlJQUunfvzrFjx0hPT6dRo0ZeG6WBAwfSt29fGjRoQJUqVZgzZw6GYRhnG8Gc/PzRE3hbVb3zgCLyJXAlzrpgQKhqH5/2/YHWuSW+siVDSZzQNde+Z8+e7bd89erVfstzes9u69atWcq6d+9O9+7dvcejRo1i1KhRGeqcf/75rFq1ynscGxtLiRLOf
```

Comment:
My first thought about the structure of the notebook was: "Interesting, why ThresholdOptimizer and GridSearch, but not ExponentiatedGradient?"
At this point this has shifted a little bit more toward "Should we be explaining why we'd use one or the other?"
To me as a dev on this project it's quite clear why, but it would be awesome if a less educated reader got that background.

Reply:
Same as above. I tried what @MiroDudik first suggested, but am happy to add more methods. Hope he can comment on this as well.

Reply:
As a dev on the project, it's *not* so clear to me why not ExpGrad - beyond the deterministic vs non-deterministic issue. Have I missed something else?

Reply:
:D My assumption was that it's about showing the tradeoff curve using grid search results, and expgrad just adds one more point on that curve (perhaps a better one than the grid points!). Perhaps not a great reason, but that's how I explained it to myself.

Let's ping @MiroDudik on this as well

Reply:
For this notebook, I'm going to just put in a brief explanation about why `GridSearch` over `ThresholdOptimizer`. We should have a more complete description in a (yet-to-be-written) user guide, which algorithm to use when... I don't think that scattering that information through notebooks makes sense.